diff --git a/sources/awesome-widget/plugin/awkeysaggregator.h b/sources/awesome-widget/plugin/awkeysaggregator.h
index f808d03..a056b3f 100644
--- a/sources/awesome-widget/plugin/awkeysaggregator.h
+++ b/sources/awesome-widget/plugin/awkeysaggregator.h
@@ -21,6 +21,7 @@
 
 #include <QHash>
 #include <QObject>
+#include <QStringList>
 
 #include "version.h"
 
diff --git a/sources/awdebug.h b/sources/awdebug.h
index 43944ce..c679281 100644
--- a/sources/awdebug.h
+++ b/sources/awdebug.h
@@ -21,9 +21,13 @@
 
 #include "version.h"
 
+#ifndef qCInfo
+#define qCInfo qCDebug
+#endif
+
 #ifndef LOG_FORMAT
 #define LOG_FORMAT                                                             \
-    "[%{time process}][%{if-debug}DD%{endif}%{if-info}II%{endif}%{if-"         \
+    "[%{time process}][%{if-debug}DD%{endif}%{if-"         \
     "warning}WW%{endif}%{if-critical}CC%{endif}%{if-fatal}FF%{endif}][%{"      \
     "category}][%{function}] %{message}"
 #endif /* LOG_FORMAT */
