diff --git a/sources/awdebug.h b/sources/awdebug.h
index 8447c65..9f8e298 100644
--- a/sources/awdebug.h
+++ b/sources/awdebug.h
@@ -23,10 +23,14 @@
 
 #include "version.h"
 
+#ifndef qCInfo
+#define qCInfo qCDebug
+#endif
+
 namespace AWDebug
 {
 const char LOG_FORMAT[] = "[%{time "
-                          "process}][%{if-debug}DD%{endif}%{if-info}II%{endif}%"
+                          "process}][%{if-debug}DD%{endif}%"
                           "{if-warning}WW%{endif}%{if-critical}CC%{endif}%{if-"
                           "fatal}FF%{endif}][%{category}][%{function}] "
                           "%{message}";
diff --git a/sources/awesome-widget/plugin/awkeysaggregator.h b/sources/awesome-widget/plugin/awkeysaggregator.h
index d29672e..f15a729 100644
--- a/sources/awesome-widget/plugin/awkeysaggregator.h
+++ b/sources/awesome-widget/plugin/awkeysaggregator.h
@@ -21,6 +21,7 @@
 
 #include <QHash>
 #include <QObject>
+#include <QStringList>
 
 #include "version.h"
 
