diff --git a/sources/awesome-widget/plugin/awkeysaggregator.h b/sources/awesome-widget/plugin/awkeysaggregator.h
index f808d03..a056b3f 100644
--- a/sources/awesome-widget/plugin/awkeysaggregator.h
+++ b/sources/awesome-widget/plugin/awkeysaggregator.h
@@ -21,6 +21,7 @@
 
 #include <QHash>
 #include <QObject>
+#include <QStringList>
 
 #include "version.h"
 
diff --git a/sources/awdebug.h b/sources/awdebug.h
index 48dc580..530c0d6 100644
--- a/sources/awdebug.h
+++ b/sources/awdebug.h
@@ -23,9 +23,9 @@
 
 #ifndef LOG_FORMAT
 #define LOG_FORMAT                                                             \
-    "[%{time process}][%{if-debug}DD%{endif}%{if-info}II%{endif}%{if-"         \
-    "warning}WW%{endif}%{if-critical}CC%{endif}%{if-fatal}FF%{endif}][%{"      \
-    "category}][%{function}] %{message}"
+    "[%{time process}][%{if-debug}DD%{endif}%{if-warning}WW%{endif}%{if-"      \
+    "critical}CC%{endif}%{if-fatal}FF%{endif}][%{category}][%{function}] "     \
+    "%{message}"
 #endif /* LOG_FORMAT */
 
 // redefine info because it doesn't log properly
